### PR TITLE
feat(DMS): update automatic topic for kafka instance

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  enable or disable automatic topic
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dms/ TESTARGS='-run TestAccKafkaInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccKafkaInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstance_basic
=== PAUSE TestAccKafkaInstance_basic
=== CONT  TestAccKafkaInstance_basic
--- PASS: TestAccKafkaInstance_basic (1507.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       1507.317s

```

### CODE REVIEW

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 1]
```diff
@@ -1355,3 +1394,80 @@ func handleMultiOperationsError(err error) (bool, error) {
...
+		var paramsJons interface{}
+		if err := json.Unmarshal(paramsData, &paramsJons); err != nil {
+			return nil, fmt.Errorf("unmarshal the response body failed: %s", err)
+		}
+
+		if topicConfig := filterAutoTopicConfig(d, paramsJons); topicConfig != nil {
+			return task, nil
+		}
+	}
+	return nil, nil
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]:
```suggestion
	params := utils.PathSearch(fmt.Sprintf("tasks|[?name=='%s'].params", taskName), resp, nil)
if params == nil{
   return  nil, fmt.Errorf("no task in the list of the automatic topic tasks")
}
paramsData := []byte(params)
var paramsJons interface{}
if err := json.Unmarshal(paramsData, &paramsJons); err != nil {
	return nil, fmt.Errorf("unmarshal the response body failed: %s", err)
}

if topicConfig := filterAutoTopicConfig(d, paramsJons); topicConfig != nil {
	return task, nil
}
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 2]
```diff
@@ -1355,3 +1394,80 @@ func handleMultiOperationsError(err error) (bool, error) {
...
+}
+
+func filterAutoTopicConfig(d *schema.ResourceData, paramsJons interface{}) interface{} {
+	rawEnableAutoTopic := d.Get("enable_auto_topic").(bool)
+	configParam := utils.PathSearch("new_kafka_configs", paramsJons, nil).(string)
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]:
```suggestion
	configParam := utils.PathSearch("new_kafka_configs", paramsJons, "").(string)
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 3]
```diff
@@ -1355,3 +1394,80 @@ func handleMultiOperationsError(err error) (bool, error) {
...
+	parts := strings.Split(configParam, ":")
+	if len(parts) < 2 {
+		return nil
+	}
+	configKey := strings.ReplaceAll(parts[0], "[", "")
+	configValue := strings.ReplaceAll(parts[1], "]", "")
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]:
```suggestion
	configKey := parts[0][1:]
	configValue := parts[1][:len(parts[1])-1]
```

Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 2 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 1 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: done

**Pull request merged**